src/Data/CategoryManager.cpp

```diff
@@ -105,8 +105,19 @@ namespace Data
 		if (keywordForm) {
 			auto hasKeyword = [keywordForm](std::string_view keywordString)
 			{
-				const auto keyword = RE::TESForm::LookupByEditorID<RE::BGSKeyword>(keywordString);
-				return keyword && keywordForm->HasKeyword(keyword);
+				const auto keywordA = RE::TESForm::LookupByEditorID<RE::BGSKeyword>(keywordString);
+				if (keywordA) {
+					return keywordForm->HasKeyword(keyword);
+				}
+				if (static auto tesDataHandler = RE::TESDataHandler::GetSingleton(); tesDataHandler) {
+					auto& keywordArray = tesDataHandler->GetFormArray<RE::BGSKeyword>();
+					for (auto& keywordB : keywordArray) {
+						if (::_stricmp(keywordB->formEditorID.data(), keywordString.data()) == 0) {
+							return keywordForm->HasKeyword(keywordB);
+						}
+					}
+				}
+				return false;
```

Comment:
If we have to query the keywords from TESDataHandler, it would be better to build a cache of all keywords (e.g. a hashtable) so they can be looked up fast.

However, this would be simple and probably good enough:
```suggestion
				const auto keywords = std::span(keywordForm->keywords, keywordForm->numKeywords);
				return std::ranges::any_of(
					keywords,
					[&](auto&& keyword)
					{
						return keyword->formEditorID.size() == keywordString.size() &&
							::_strnicmp(
								keyword->formEditorID.data(),
								keywordString.data(),
								keywordString.size()) == 0;
					});
```

Reply:
I can confirm this works nice and fast, thank you.